homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -162,8 +164,12 @@ def parsestatus(self, status):
     def update(self):
         """Update the alarm status."""
         if not self._rs_enabled:
```

Comment:
Instead, move this line to the `should_poll` property that you can define and have it return `False` if `self._rs_enabled` is true otherwise return `True`.